src/common/util/include/openvino/util/ov_filesystem.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#if !(defined(_MSC_VER) && __cplusplus == 199711L)
+#    if __cplusplus >= 201103L
+#        define CPP_VER_11
+#        if __cplusplus >= 201402L
+#            define CPP_VER_14
+#            if __cplusplus >= 201703L
+#                define CPP_VER_17
+#                if __cplusplus >= 202002L
+#                    define CPP_VER_20
+#                endif
+#            endif
+#        endif
+#    endif
+#elif defined(_MSC_VER) && __cplusplus == 199711L
+#    if _MSVC_LANG >= 201103L
+#        define CPP_VER_11
+#        if _MSVC_LANG >= 201402L
+#            define CPP_VER_14
+#            if _MSVC_LANG >= 201703L
+#                define CPP_VER_17
+#                if _MSVC_LANG >= 202002L
+#                    define CPP_VER_20
+#                endif
+#            endif
+#        endif
+#    endif
+#endif
```

Comment:
@praasz do you think we can move these C++ standard declarations to common header https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/core/visibility.hpp#L70 ? 

I think in future it can be useful for the whole library.

Reply:
Yes, these macros can be usefull. Lets move them to visibilit.
And add prefix them with `OV` or `OPENVINO` like `OPENVINO_CPP_VER_20`.

Reply:
`OPENVINO_CPP_VER_20` looks good

Reply:
Done